### DISCUSSION TITLE
Trying to convert *.cr2 to *.jpeg, but colors became different

### DISCUSSION THREAD

**Discussion by ydmitry** [dlemstra/magick-wasm#26] [Q&A]
Hello,

Trying to use `magic-wasm` with code like this:

```ts
private process(file: File): Promise<File> {
    return new Promise((resolve, reject) => {
      import('@imagemagick/magick-wasm').then(async ({ ImageMagick, initializeImageMagick }) => {
        await initializeImageMagick();
        ImageMagick.read(new Uint8Array(await fileToArrayBuffer(file)), (image) => {
          image.write((data) => {
            resolve(
              new File([data], 'output.jpg', {
                type: 'image/jpeg',
                lastModified: Date.now(),
              }),
            );
          }, MagickFormat.Jpeg);
        });
      });
    });
  }
```

And trying to upload Canon Raw file, in the result I see JPEG, but with more yellow tones. Maybe I can pass some config to fix this? 

Before (CR2 file):
<img width="1043" alt="Screenshot 2021-09-19 at 18 18 23" src="https://user-images.githubusercontent.com/1408154/133933081-48c19a03-b7d6-4909-86fc-6835121eb8cc.png">

After (JPEG):
<img width="1047" alt="Screenshot 2021-09-19 at 18 18 13" src="https://user-images.githubusercontent.com/1408154/133933071-d86874ae-97e5-42dd-a5a5-118c61cd35a7.png">

**Comment by dlemstra** [comment]
Have you tried to use the `DngReadDefines` class? Not sure which settings you need to set but you can probably get a different output with one of these settings.

**Reply by ydmitry** [reply]
Didn't try, will it help not only with DNG format, but with CR2 too? Don't know exactly which settings need to pass, but according to dicussions in imagick forum (https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=6936) it can be:
```
convert cr2:IMG_9459.CR2 image.png
```
or
```
convert E1DXINBI003200.CR2 -set colorspace RGB -resize 800x600 -colorspace sRGB out.png
```

Didn't try which will work because don't know how to specify this with interface of wasm library...

**Reply by dlemstra** [reply]
You can set the `DngReadDefines` with the code example below:

```js
const dngReadDefines = new DngReadDefines();
dngReadDefines.useCameraWhitebalance = true; // Not sure which setting you will need.

const readSettings = new MagickReadSettings();
readSettings.setDefines(dngReadDefines);

ImageMagick.read(new Uint8Array(await fileToArrayBuffer(file)), readSettings, (image) => {
}
```

**Reply by ydmitry** [reply]
@dlemstra where can I find available settings?

**Reply by dlemstra** [reply]
I am not sure what you are asking, Do you mean the file that contains those settings? That is `formats/dng/dng-read-defines.js` and `settings/magick-read-settings.js`.
